File: examples/benchmarks/fib.js
#!/usr/bin/env node

function fib(n) {
	return n < 2 ? n : fib(n-2) + fib(n-1);
}

n = 20
if(process.argv[2]) {
	n = parseInt(process.argv[2]);
}

console.log(fib(n));
process.exit();

File: examples/benchmarks/fib.go
#!/usr/bin/env jit.go

package main;

import "os"
import "fmt"
import "strconv"

func fib(n int) int {
	if n < 2 {
   	 return n
	}
	
	return fib(n-2) + fib(n-1)
}

func main() {
	var err error
	var n int
	if len(os.Args) >= 2 {
		n, err = strconv.Atoi(os.Args[1])
		if err != nil {
			n = 0
		}
	}
	if n == 0 { 
		n = 20 
	}
	
	fmt.Fprintf(os.Stdout, "%d\n", fib(n))
}
